parser-native/src/main/java/ru/urururu/sanity/cpp/NativeInstructionParser.java

```diff
@@ -0,0 +1,478 @@
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMLoad;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return null;
+        }
+
+        @Override
+        public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return createLoad(ctx, bitreader.LLVMGetOperand(instruction, 0));
+        }
+    }
+
+    private class RetParser extends AbstractParser {
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMRet;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            if (bitreader.LLVMGetNumOperands(instruction) == 0) {
+                return createReturn(ctx, instruction);
+            }
+
+            return createReturn(ctx, instruction, bitreader.LLVMGetOperand(instruction, 0));
+        }
+    }
+
+    private static class AllocaParser extends AbstractParser {
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMAlloca;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return null;
+        }
+
+        @Override
+        public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return ctx.getOrCreateTmpVar(instruction);
+        }
+    }
+
+    private class GetElementPtrParser extends AbstractParser {
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMGetElementPtr;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return null;
+        }
+
+        @Override
+        public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            int operandsCount = bitreader.LLVMGetNumOperands(instruction);
+
+            return getPointer(ctx, bitreader.LLVMGetOperand(instruction, 0), Iterables.indexed(
+                    i -> bitreader.LLVMGetOperand(instruction, i + 1),
+                    () -> operandsCount - 1));
+        }
+
+        @Override
+        public RValue parseConst(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue constant) {
+            return parseValue(ctx, constant);
+        }
+    }
+
+    private class ExtractValueParser extends AbstractParser {
+        @Override
+        public LLVMOpcode getOpcode() {
+            return LLVMOpcode.LLVMExtractValue;
+        }
+
+        @Override
+        public Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            return null;
+        }
+
+        @Override
+        public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+            // todo wtf? tests?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)